### PR TITLE
10/Add independent demo-project that uses mavenLocal() for testing

### PR DESCRIPTION
**Pull request by friedger**
This PR adds a demo project that can be used for testing when the bintray plugin is uploaded in dry run mode.

You find the results of ./gradlew bintrayUpload in your local .m2 folder
screen shot 2015-04-02 at 19 23 51

### CODE REVIEW

**Comment by ouchadam** [comment]
not sure if this is necessary as the plugin source can release itself.

Maybe adding the `mavenLocal()` to the core would be simpler but then since it's a one line change it should probably be down to the dev

File: demo-project/build.gradle [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    repositories {
+        jcenter()
+        mavenLocal()
```
Comment (ouchadam) [inline demo-project/build.gradle]: the .m2 is the default output for dryRun=true, this isn't needed!

Comment (ouchadam) [inline demo-project/build.gradle]: ignore that, I see what it's doing now :+1:

**Comment by blundell** [comment]
I think this adds more confusion than it serves. What do we gain by having this demo?

**Comment by friedger** [comment]
It helps to test/develop the plugin. Eventually, this will be part of some unit tests and so on.
I would even add a buildSrc folder.

**Comment by xrigau** [comment]
I think this is good :+1: let's :shipit: and if it doesn't help us we can revert it at any point

**Comment by xrigau** [comment]
will wait a few minutes before merge to see if someone has more comments

**Comment by blundell** [comment]
wait

**Comment by blundell** [comment]
demo-project module should have it's own `README` explaining what it does, what it is for :x:

**Comment by blundell** [comment]
The benefit of this @ouchadam is you can say `depends compile('library');` so that you can test the lastest plugin version without having to release it

**Comment by ouchadam** [comment]
nope can't do that since it's a buildscript plugin, we'd have to symlink a `buildSrc` dir in the demo project to `core/src`

**Comment by blundell** [comment]
ah then I understand why this "sand pit" isn't working atm

**Comment by ouchadam** [comment]
yeah this is my conflict, somewhere for us to simply deploy and test the plugin is 100% needed.

Maybe renaming this to sandpit/playground would help, but we would still have to do the awkward deploy dance. An acceptance suite would be super useful here

**Comment by blundell** [comment]
some type of script to do the dance for us?

**Comment by friedger** [comment]
Sqlite-analyzer does the buildSrc thing without symlink. It can be easily
added after this PR has been merged.
Am 12.04.2015 17:22 schrieb "Paul Blundell" <notifications@github.com>:

**Comment by stefanhoth** [comment]
Can we merge this as an intermediate step or is more needed? @blundell @ouchadam @xrigau

**Comment by ouchadam** [comment]
For me this needs a name change from demo and the gradle wrapper to be
removed
On 13 Apr 2015 09:10, "Stefan Hoth" <notifications@github.com> wrote:

> Can we merge this as an intermediate step or is more needed? @blundell
> <https://github.com/blundell> @ouchadam <https://github.com/ouchadam>
> @xrigau <https://github.com/xrigau>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/bintray-release/pull/30#issuecomment-92262037>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

**Comment by blundell** [comment]
Yeah I would rather this was a full feature before we merge half of something into master. 

Like Adam says name change, it also needs it's own README explaining its use and why people should want to use it or how they should use it.

Then looking at the scripting that would make this a useful test sandpit

**Pull request closed without merging**